sc-memory/sc-core/sc_memory_ext.c

```diff
@@ -68,7 +74,7 @@ sc_result sc_ext_initialize(
 #if SC_IS_PLATFORM_MAC
   gchar const * moduleSuffix = "dylib";
 #else
-  gchar const * moduleSuffix = G_MODULE_SUFFIX;
+  gchar const * moduleSuffix = "so";
```

Comment:
This is not always the case since theoretically Linux and Darwin are not the only platforms that you can compile this code for. Why do we need this change?

Reply:
G_MODULE_SUFFIX is deprecated in the glib 2.75, which can be used in ubuntu-24.04 or latest updates for macos-13 and macos-14. The main issue with G_MODULE_SUFFIX is its inability to consistently provide the correct file extension for dynamically loaded modules across different platforms. For instance, while Unix-like systems typically use .so for shared libraries, Windows uses .dll, and macOS has used both .dylib and .so.

The deprecation of G_MODULE_SUFFIX is also tied to the introduction of better alternatives that provide more flexibility and accuracy. Functions like g_module_open() are recommended, which allow specifying the module name without the extension, letting the system determine the correct suffix dynamically. This approach is more robust and reduces the risk of errors related to incorrect suffix assumptions. But these features were implemented in the last versions of glib. So, I left this logic to save compatibility with the oldest operating systems.

Reply:
I'd leave a `TODO:` here to refactor it later to use the modern alternative

Reply:
I removed this logic. It works without it on ubuntu-20.04